spring-cloud-dataflow-module-registry/src/main/java/org/springframework/cloud/dataflow/module/registry/ArtifactRegistry.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.module.registry;
+
+
+import java.util.List;
+
+import org.springframework.cloud.dataflow.core.ArtifactType;
+
+/**
+ * {@code ArtifactRegistry} is used to manage module registrations, as well
+ * as supporting libraries. This includes operations such as find, register, and delete.
+ *
+ * @see ArtifactRegistration
+ *
+ * @author Eric Bottard
+ * @author David Turanski
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ */
+public interface ArtifactRegistry {
+
+	/**
+	 * Look up the registration for an artifact by name and type. If
+	 * a registration does not exist, {@code null} is returned.
+	 *
+	 * @param name the artifact name
+	 * @param type the artifact type
+	 *
+	 * @return registration for a module/library, or {@code null} if not found
```

Comment:
just replace "module/library" with "artifact" in this class... it's meant to be completely generic at this level